apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue

```diff
@@ -21,39 +21,45 @@
 
 <script lang="ts" setup>
 import {computed, toRef, toValue} from 'vue';
-import {format} from 'date-fns';
+import {format, setDay} from 'date-fns';
 import {OPENING_HOURS, CLOSED} from '@myparcel-do/shared';
-import {useLanguage, usePickupLocation} from '../../../../composables';
+import {useLanguage, usePickupLocation, useDateFormat} from '../../../../composables';
+
+/**
+ * Returns the localized name of a weekday for a given index (0 = Sunday, 1 = Monday, ...).
+ */
+const getLocalizedWeekdayName = (weekdayIndex: number): string => {
+  const date = setDay(new Date(), weekdayIndex, {weekStartsOn: 0});
+  const name = useDateFormat(date).weekday;
+  return name.value;
+};
 
 const props = defineProps<{locationCode: string; expanded?: boolean}>();
 const {translate} = useLanguage();
 const locationCodeRef = toRef(props, 'locationCode');
 const {pickupLocation} = usePickupLocation(locationCodeRef);
 
-const weekDaysMap = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
-
 const openingHours = computed(() => {
   const openingHoursData = toValue(pickupLocation)?.openingHours ?? [];
-  return weekDaysMap.map((day, index) => {
-    const dayData = openingHoursData.find((entry: {weekday: number}) => entry.weekday === index);
-
-    if (dayData?.hours && dayData.hours.length > 0) {
-      const times = dayData.hours[0];
-
-      if (times.start && times.end) {
-        const startTime = format(new Date(times.start.date), 'HH:mm');
-        const endTime = format(new Date(times.end.date), 'HH:mm');
-        const timeString = `${startTime} - ${endTime}`;
-        return {
-          weekday: day,
-          timeString,
-        };
+  return Array.from({length: 7}).map((_, weekdayIndex) => {
```

Comment:
We are encouraged not to use ‘magic numbers’ but in this case I suppose it is pretty clear that 7 means the number of days in a week.